### PR TITLE
feat(component): add tab trap for modals and dialogs

### LINKED ISSUE #252: Trap tab key on modals
**Is your feature request related to a problem? Please describe.**
For accessibility, we need to trap the tab key for Modals/Dialogs.

**Describe the solution, feature, or improvement you'd like**
Rolling your own functionality for this can be quite cumbersome, so we'll probably want to use a library for it.

### PR DESCRIPTION
**Pull request by chanceaclark**
## What
trap

Add Tab trapping to dialogs and modals. Closes #252.

`focus-trap` is about 2kb (gzip + min) so it shouldn't add too much on the bundle size of BigDesign: https://bundlephobia.com/result?p=focus-trap@5.1.0

### CODE REVIEW

**Comment by jordan-massingill** [comment]
The code lgtm - easy to read and understand 👍

File: packages/big-design/src/components/Modal/Modal.tsx [hunk 1]
```diff
@@ -68,18 +71,33 @@ export class Modal extends React.PureComponent<ModalProps, ModalState> {
...
     }
 
-    this.props.isOpen
-      ? (document.body.style.overflowY = 'hidden')
-      : (document.body.style.overflowY = this.state.initialBodyOverflowY);
+    if (this.props.isOpen) {
+      document.body.style.overflowY = 'hidden';
+
+      if (this.focusTrap) {
+        this.focusTrap.activate();
```
Comment (deini) [inline packages/big-design/src/components/Modal/Modal.tsx]: 🍹 `this.focusTrap?.activate();`

File: packages/big-design/src/components/Modal/Modal.tsx [hunk 2]
```diff
@@ -68,18 +71,33 @@ export class Modal extends React.PureComponent<ModalProps, ModalState> {
     }
 
     document.body.style.overflowY = initialBodyOverflowY;
+
+    this.returnFocus();
   }
 
   componentDidUpdate(prevProps: ModalProps) {
+    if (this.modalRef.current) {
```
Comment (deini) [inline packages/big-design/src/components/Modal/Modal.tsx]: Offline review: cleanup `focusTrap` and add a `null` check for it here.

File: packages/big-design/src/components/Modal/Modal.tsx [hunk 3]
```diff
@@ -68,18 +71,33 @@ export class Modal extends React.PureComponent<ModalProps, ModalState> {
...
+      document.body.style.overflowY = 'hidden';
+
+      if (this.focusTrap) {
+        this.focusTrap.activate();
+      }
+    } else {
+      document.body.style.overflowY = this.state.initialBodyOverflowY;
+
+      if (this.focusTrap) {
+        this.focusTrap.deactivate();
```
Comment (deini) [inline packages/big-design/src/components/Modal/Modal.tsx]: 🍹 same as above

File: packages/big-design/src/components/Modal/Modal.tsx (hunk 1 above)
Comment (chanceaclark) [reply packages/big-design/src/components/Modal/Modal.tsx]: Totally forgot we could use those now 🌶

**Pull request merged**